mteb/evaluation/CacheWrapper.py

```diff
@@ -0,0 +1,261 @@
+
+    def _double_vectors_file(self):
+        current_size = len(self.vectors)
+        new_size = current_size * 2
+        logger.info(f"Doubling vectors file from {current_size} to {new_size} vectors")
+        self.vectors.flush()
+        new_vectors = np.memmap(
+            self.vectors_file,
+            dtype="float32",
+            mode="r+",
+            shape=(new_size, self.vector_dim),
+        )
+        new_vectors[:current_size] = self.vectors[:]
+        self.vectors = new_vectors
+
+    def save(self) -> None:
+        try:
+            self.vectors.flush()
+            with open(self.index_file, "wb") as f:
+                pickle.dump(self.hash_to_index, f, protocol=pickle.HIGHEST_PROTOCOL)
+            logger.info(f"Saved TextVectorMap to {self.directory}")
+        except Exception as e:
+            logger.error(f"Error saving TextVectorMap: {str(e)}")
+            raise
+
+    def load(self, name: str = None) -> None:
+        name_details = name if name else ""
+        try:
+            if self.index_file.exists() and self.vectors_file.exists():
+                with open(self.index_file, "rb") as f:
+                    self.hash_to_index = pickle.load(f)
+
+                self.vectors = np.memmap(self.vectors_file, dtype="float32", mode="r+")
+                self.vectors = self.vectors.reshape(-1, self.vector_dim)
+
+                logger.info(
+                    f"Loaded TextVectorMap ({name_details}) from {self.directory}"
+                )
+                logger.info(f"Loaded vectors file with shape: {self.vectors.shape}")
+            else:
+                logger.warning(
+                    f"No existing files found. Initialized empty TextVectorMap ({name_details})."
+                )
+        except Exception as e:
+            logger.error(f"Error loading TextVectorMap ({name_details}): {str(e)}")
+            raise
+
+    def get_vector(self, text: str) -> Optional[np.ndarray]:
+        try:
+            text_hash = self._hash_text(text)
+            if text_hash not in self.hash_to_index:
+                logger.debug(f"Text hash not found in index: {text_hash}")
+                return None
+            index = self.hash_to_index[text_hash]
+            return self.vectors[index]
+        except Exception as e:
+            logger.error(f"Error retrieving vector for text: {str(e)}")
+            raise
+
+    def __contains__(self, text: str) -> bool:
+        return self._hash_text(text) in self.hash_to_index
+
+
+class CachedEmbeddingWrapper:
+    def __init__(self, model: Any, cache_path: Union[str, Path], vector_dim: int = 768):
+        self._model = model
+        self.cache_path = Path(cache_path)
+        self.cache_path.mkdir(parents=True, exist_ok=True)
+        self.vector_dim = vector_dim
+
+        if hasattr(model, "encode_queries") and hasattr(model, "encode_corpus"):
+            self.encode_method = "split"
+            self.query_cache = TextVectorMap(
```

Comment:
Agree! I do that here: https://github.com/embeddings-benchmark/mteb/blob/89db68e943713414e2eca1680085572a8c0790e8/mteb/evaluation/MTEB.py#L378 which is then passed to this class.

Reply:
Ah, I see. But since you're recreating the model, this would result in a wrapper inside another wrapper, leading to multiple caches of the same data.

Reply:
In main branch `task_name` and `prompt_type` currently passing to model_encode and from there to `model.encode` in kwargs in `prompt_name`

Reply:
Thanks @Samoed for pointing out this issue.  I've moved it out to avoid this issue!